// @google-cloud/dataflow/build/src/index.js
"use strict";
// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by synthtool. **
// ** https://github.com/googleapis/synthtool **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.protos = exports.TemplatesServiceClient = exports.SnapshotsV1Beta3Client = exports.MetricsV1Beta3Client = exports.MessagesV1Beta3Client = exports.JobsV1Beta3Client = exports.FlexTemplatesServiceClient = exports.v1beta3 = void 0;
const v1beta3 = require("./v1beta3");
exports.v1beta3 = v1beta3;
const FlexTemplatesServiceClient = v1beta3.FlexTemplatesServiceClient;
exports.FlexTemplatesServiceClient = FlexTemplatesServiceClient;
const JobsV1Beta3Client = v1beta3.JobsV1Beta3Client;
exports.JobsV1Beta3Client = JobsV1Beta3Client;
const MessagesV1Beta3Client = v1beta3.MessagesV1Beta3Client;
exports.MessagesV1Beta3Client = MessagesV1Beta3Client;
const MetricsV1Beta3Client = v1beta3.MetricsV1Beta3Client;
exports.MetricsV1Beta3Client = MetricsV1Beta3Client;
const SnapshotsV1Beta3Client = v1beta3.SnapshotsV1Beta3Client;
exports.SnapshotsV1Beta3Client = SnapshotsV1Beta3Client;
const TemplatesServiceClient = v1beta3.TemplatesServiceClient;
exports.TemplatesServiceClient = TemplatesServiceClient;
exports.default = {
    v1beta3,
    FlexTemplatesServiceClient,
    JobsV1Beta3Client,
    MessagesV1Beta3Client,
    MetricsV1Beta3Client,
    SnapshotsV1Beta3Client,
    TemplatesServiceClient,
};
const protos = require("../protos/protos");
exports.protos = protos;
