// @google-cloud/dataflow/build/src/v1beta3/index.js
"use strict";
// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by gapic-generator-typescript. **
// ** https://github.com/googleapis/gapic-generator-typescript **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesServiceClient = exports.SnapshotsV1Beta3Client = exports.MetricsV1Beta3Client = exports.MessagesV1Beta3Client = exports.JobsV1Beta3Client = exports.FlexTemplatesServiceClient = void 0;
var flex_templates_service_client_1 = require("./flex_templates_service_client");
Object.defineProperty(exports, "FlexTemplatesServiceClient", { enumerable: true, get: function () { return flex_templates_service_client_1.FlexTemplatesServiceClient; } });
var jobs_v1_beta3_client_1 = require("./jobs_v1_beta3_client");
Object.defineProperty(exports, "JobsV1Beta3Client", { enumerable: true, get: function () { return jobs_v1_beta3_client_1.JobsV1Beta3Client; } });
var messages_v1_beta3_client_1 = require("./messages_v1_beta3_client");
Object.defineProperty(exports, "MessagesV1Beta3Client", { enumerable: true, get: function () { return messages_v1_beta3_client_1.MessagesV1Beta3Client; } });
var metrics_v1_beta3_client_1 = require("./metrics_v1_beta3_client");
Object.defineProperty(exports, "MetricsV1Beta3Client", { enumerable: true, get: function () { return metrics_v1_beta3_client_1.MetricsV1Beta3Client; } });
var snapshots_v1_beta3_client_1 = require("./snapshots_v1_beta3_client");
Object.defineProperty(exports, "SnapshotsV1Beta3Client", { enumerable: true, get: function () { return snapshots_v1_beta3_client_1.SnapshotsV1Beta3Client; } });
var templates_service_client_1 = require("./templates_service_client");
Object.defineProperty(exports, "TemplatesServiceClient", { enumerable: true, get: function () { return templates_service_client_1.TemplatesServiceClient; } });
